packages/neuron-wallet/src/services/rpc-service.ts

```diff
@@ -29,6 +30,23 @@ export default class RpcService {
     return BlockHeader.fromSDK(result)
   }
 
+  public async getTransactionIncludeRejected(
+    hash: string
+  ): Promise<TransactionWithStatus | undefined | { transaction: null; txStatus: TxStatus }> {
+    const result = await this.rpc.getTransaction(hash)
+    if (result?.transaction) {
+      return TransactionWithStatus.fromSDK(result)
+    }
+    if (result.txStatus.status === TxStatusType.Rejected) {
+      logger.warn(`Transaction[${hash}] was rejected`)
+      return {
+        transaction: null,
+        txStatus: TxStatus.fromSDK(result.txStatus),
+      }
+    }
+    return undefined
```

Comment:
Is the branch `if (result.txStatus.status === TxStatusType.Rejected)` reachable? If a transaction is rejected, the `result.transaction` is not nil but with a `status: rejected`. It goes into the `if(result?.transaction)` branch no matter if it is rejected

Reply:
> Is the branch `if (result.txStatus.status === TxStatusType.Rejected)` reachable? If a transaction is rejected, the `result.transaction` is not nil but with a `status: rejected`. It goes into the `if(result?.transaction)` branch no matter if it is rejected

A rejected transaction: 

```
{
  transaction: null,
  txStatus: {
    blockHash: null,
    status: 'rejected',
    reason:
      '{"type":"RBFRejected","description":"RBF rejected: replaced by tx Byte32(0xccf7d9b2f739d5b366235c5a17b2d55bb423d3e4ebbbaa6dd446bb1c84c36a60)"}',
  },
  timeAddedToPool: null,
  cycles: null,
  fee: null,
  min_replace_fee: null,
}
```